src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/controllers/RestrictedPatientsController.kt

```diff
@@ -12,33 +13,68 @@ import org.springframework.web.bind.annotation.RestController
 import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.request.DischargeToHospitalRequest
 import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.request.MigrateInRequest
 import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.response.RestrictedPatientDto
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.services.DomainEventPublisher
 import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.services.RestrictedPatientsService
 
 @RestController
-class RestrictedPatientsController(private val restrictedPatientsService: RestrictedPatientsService) {
+class RestrictedPatientsController(
+  private val restrictedPatientsService: RestrictedPatientsService,
+  private val domainEventPublisher: DomainEventPublisher,
+  private val telemetryClient: TelemetryClient,
+) {
   @PostMapping(
     value = ["/discharge-to-hospital"],
     consumes = [MediaType.APPLICATION_JSON_VALUE],
   )
   @ResponseStatus(code = HttpStatus.CREATED)
   fun dischargeToHospital(@RequestBody dischargeToHospital: DischargeToHospitalRequest): RestrictedPatientDto =
-    restrictedPatientsService.dischargeToHospital(dischargeToHospital)
+    restrictedPatientsService.dischargeToHospital(dischargeToHospital).also {
+      it.publishAndTrackAdd()
+    }
 
   @PostMapping(
     value = ["/migrate-in-restricted-patient"],
     consumes = [MediaType.APPLICATION_JSON_VALUE],
   )
   @ResponseStatus(code = HttpStatus.CREATED)
   fun migrateInPatient(@RequestBody migrateIn: MigrateInRequest): RestrictedPatientDto =
-    restrictedPatientsService.migrateInPatient(migrateIn)
+    restrictedPatientsService.migrateInPatient(migrateIn).also {
+      it.publishAndTrackAdd()
+    }
 
-  @GetMapping(
-    value = ["/restricted-patient/prison-number/{prison-number}"],
-  )
+  private fun RestrictedPatientDto.publishAndTrackAdd() {
+    domainEventPublisher.publishRestrictedPatientAdded(prisonerNumber)
```

Comment:
we now publish events *after* the service call and therefore after the transaction has been completed / committed. This means that the indexer can listen to the events and then get the correct state

Reply:
Is there a test that proves the transaction has been comitted before the domain event has been published?